### PR TITLE
User interface: admin pages and jinja templates

### PR DESCRIPTION
**Pull request by jma**
* BETTER: lighten the front page search input button color
* BETTER: resize the front page background image
* BETTER: remove unused SCSS rules or replace them by a bootstrap class
* BETTER: enhance the search results views
* BETTER: enhance the editors views
* NEW: add delete fonction in libraries and documents brief views
* FIX: fix locale for schema and forms
* add a rule for typscript files in the .editorconfig

Co-authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>
Signed-off-by: Igor Milhit <igor.milhit@rero.ch>

--------

# To test

- bootstrap
- setup
- display admin pages, frontpage, document detailed view...

### CODE REVIEW

**Comment by Garfield-fr** [comment]
On the list, the arrow on the right on the connected records is not visible. It must be put before the text on the left.

**Comment by Garfield-fr** [comment]
I don't know if it's right to put an "add" button for a item on the list. I think it should be in the document.

File: ui/admin/src/app/records/search/brief-view/bio-informations.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'bioInformations'
+})
+export class BioInformationsPipe implements PipeTransform {
+
+  transform(value: any, args?: any): any {
```
Comment (Garfield-fr) [inline ui/admin/src/app/records/search/brief-view/bio-informations.pipe.ts]: args is not used in function ?

File: ui/admin/src/app/records/search/brief-view/birth-date.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'birthDate'
+})
+export class BirthDatePipe implements PipeTransform {
+
+  transform(value: any, args?: any): any {
```
Comment (Garfield-fr) [inline ui/admin/src/app/records/search/brief-view/birth-date.pipe.ts]: args is not used in function ?

File: ui/admin/src/app/records/search/brief-view/libraries-brief-view.component.ts [hunk 3]
```diff
@@ -1,16 +1,127 @@
...
+    ) {}
+
+  toggleCollapse() {
+    if (this.isCollapsed) {
+      console.log('loading data');
+      const libraryPid = this.record.metadata.pid;
+      this.recordsService
+          .getRecords('locations', 1, 100, `library.pid:${libraryPid}`)
+          .subscribe(data => {
+            console.log('recieved', data);
```
Comment (Garfield-fr) [inline ui/admin/src/app/records/search/brief-view/libraries-brief-view.component.ts]: To be deleted

File: ui/admin/src/app/records/search/brief-view/libraries-brief-view.component.ts [hunk 4]
```diff
@@ -1,16 +1,127 @@
...
+    } else {
+      this.isCollapsed = !this.isCollapsed;
+    }
+  }
+
+  deleteLocation(pid) {
+    this.recordsService.delete('locations', pid).subscribe(locations => {
+      this.locations = this.locations.filter(location => location.metadata.pid !== pid);
+    });
+    console.log(pid);
```
Comment (Garfield-fr) [inline ui/admin/src/app/records/search/brief-view/libraries-brief-view.component.ts]: To be deleted

**Comment by Garfield-fr** [comment]
In the item view, there is a json print.

**Comment by Garfield-fr** [comment]
In the list of libraries, the arrow is on the left when the information is closed, but on the right when it is open.

File: ui/admin/src/app/records/search/brief-view/bio-informations.pipe.ts (hunk 1 above)
Comment (jma) [reply ui/admin/src/app/records/search/brief-view/bio-informations.pipe.ts]: no this has been generated by angular tools, I will remove it.

**Comment by jma** [comment]
@Garfield-fr I was unable to reproduce the arrow position in the libraries view nor seen the json in the item view.

File: ui/admin/src/app/records/search/brief-view/bio-informations.pipe.ts (hunk 1 above)
Comment (Garfield-fr) [reply ui/admin/src/app/records/search/brief-view/bio-informations.pipe.ts]: not deleted ?

File: ui/admin/src/app/records/editor/editor.component.ts [hunk 5]
```diff
@@ -75,6 +80,10 @@ export class EditorComponent implements OnInit {
             this.schemaForm.schema.properties.document
               .properties['$ref']['default'] = urlPerfix + query.document;
           }
+          if (this.recordType === 'locations' && query.library) {
+            this.redirectRecordType = 'libraries';
+            this.schemaForm.schema.properties.library.properties['$ref']['default'] = 'http://ils.rero.ch/api/libraries/' + query.library;
```
Comment (Garfield-fr) [inline ui/admin/src/app/records/editor/editor.component.ts]: Use ApiService ton generate the url

**Pull request merged**